monai/data/image_reader.py

```diff
@@ -57,6 +57,13 @@
 cp, has_cp = optional_import("cupy")
 kvikio, has_kvikio = optional_import("kvikio")
 
+if TYPE_CHECKING:
+    import cupy
+
+    NdarrayOrCupy = Union[np.ndarray, cupy.ndarray]
```

Comment:
This was done to get around Cupy being an optional dependency, this makes it necessary when type checking but this might not be avoidable right now. We do need to consider a way of defining placeholder types for optional dependencies so we can type check correctly without these workarounds.